apps/aecore/lib/aecore/wallet/worker.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Aecore.Wallet.Worker do
+  @defmodule """
+  Module for handling the creation of a Wallet file
+  """
+
+  use GenServer
+
+  @typedoc "Options for network"
+  @type opts :: :mainnet | :testnet
+
+  @aewallet_dir Application.get_env(:aecore, :aewallet)[:path]
+  @aewallet_pass "1234"
+
+  ## Client API
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{path: @aewallet_dir}, name: __MODULE__)
+  end
+
+  def init(%{path: path} = state) do
+    has_wallet(File.exists?(path))
+    {:ok, state}
+  end
+
+  @spec get_public_key(String.t()) :: binary()
+  def get_public_key(password) do
+    GenServer.call(__MODULE__, {:get_pub_key, {password, :mainnet}})
+  end
+
+  @spec get_public_key(String.t(), opts()) :: binary()
+  def get_public_key(password, network) do
```

Comment:
Do we need to protect the public key with a password?

Reply:
The password is used to decrypt the wallet. Everytime we derive some key, we need to decrypt the wallet.